src/client.js

```diff
@@ -295,13 +295,14 @@ Client.prototype.connect = function(args) {
 	client.save();
 };
 
-Client.prototype.generateToken = function(callback) {
-	crypto.randomBytes(48, (err, buf) => {
-		if (err) {
-			throw err;
-		}
-
-		callback(buf.toString("hex"));
+Client.prototype.generateToken = function() {
+	return new Promise((resolve, reject) => {
+		crypto.randomBytes(48, (err, buf) => {
+			if (err) {
+				reject(err);
```

Comment:
You're going to reject and then resolve it, you need a `return` in here.

Reply:
I'm pretty sure once it's rejected it can't be resolved anymore, but fixed anyway.